chore: simplify RFC format (#222)
aws/aws-cdk-rfcs: 0000-template.md

---

> rfc pr: https://github.com/aws/aws-cdk-rfcs/pull/xxx <-- fill this after you've already created the PR

Reviewer: nit: maybe something like this so it looks clean?

```suggestion
rfc pr: [#xxx](https://github.com/aws/aws-cdk-rfcs/pull/xxx) <-- fill this after you've already created the PR
```

> related issue: https://github.com/aws/aws-cdk-rfcs/issues/yyy

Reviewer: ```suggestion
tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/yyy
```

---

# [TITLE]

> One sentence: brief description of the feature from a user perspective.

## Working Backwards

> This section should contain one or more "artifacts from the future", as if the
> feature was already released and we are publishing its CHANGELOG, README
> and optionally a PRESS RELEASE.
>
> This is the most important section of your RFC. It's a powerful thought
> exercise which will challenge you to truly think about this feature from a
> user's point of view.

### CHANGELOG

> Write the changelog entry for this feature in conventional form (e.g.
> `feat(eks): cluster tags`).
>
> If this change includes a breaking change, include a `BREAKING CHANGE` clause
> with information on how to migrate. If migration is complicated, refer to a
> fictional GitHub issue and add its contents here.

> ### README

Reviewer 2: Do we need the `README` section in addition to the content of the `PRESS RELEASE` and `Working Backwards`,  I feel like there will be a lot of duplicate between the two.

Author: See line 12, the requirement is to include "one or more artifacts", not all of them.

> If this is a new feature, write the README section which describes this new
> feature. It should describe the feature and walk users through usage examples
> and description of the various options and behavior.

### PRESS RELEASE

> > If this is a major feature (~6 months of work), write the PRESS RELEASE which

Reviewer 3: This is covered in Working Backwards above. We should merge these paragraphs.

Author: This is the working backwards section. Not sure I follow...

> any known patterns or best practices the design is based around.
> If this is a new feature, write the README section which describes this new
> feature. It should describe the feature and walk users through usage examples
> and description of the various options and behavior.

# Detailed Design
### PRESS RELEASE

> This is the bulk of the RFC. Explain the design in enough detail for somebody
> familiar with CDK to understand, and for somebody familiar with the
> implementation to implement. This should get into specifics and corner-cases,
> and include examples of how the feature is used. Any new terminology should be
> defined here.
>
> Include any diagrams and/or visualizations that help to demonstrate the design.
> Here are some tools that we often use:
>
> - [Graphviz](http://graphviz.it/#/gallery/structs.gv)
> - [PlantText](https://www.planttext.com)
> If this is a major feature (~6 months of work), write the PRESS RELEASE which
> announces this feature. The press release is a single page that includes 7
> paragraphs: (1) summary, (2) problem, (3) solution, (4) leader quote, (5) user
> experience, (6) customer testimonial and (7) one sentence call to action.

> # Drawbacks

Reviewer 3: I think this was an important section.

Author: it's in the FAQ

Author: added

> > This section includes a answers to questions customers will likely ask about

Reviewer: ```suggestion
> This section should contain answers to questions customers will likely ask about
```

### Why should we _not_ do this?

> Is there a way to address this use case with the current product? What are the
> downsides of implementing this feature?

### What changes are required to enable this change?

> Briefly describe the high-level design approach for implementing this feature.
>
> As appropriate, you can add an appendix with a more detailed design document.
>
> This is a good place to reference a prototype or proof of concept, which is
> highly recommended for most RFCs.

### Is this a breaking change?

> If the answer is no. Otherwise:
>
> Describe what ways did you consider to deliver this without breaking users?
>
> Make sure to include a `BREAKING CHANGE` section above with a a description of the breaking
> changes and the migration path.

### What alternative solutions did you consider?

> > Briefly describe alternative approaches that you considered.

Reviewer: Brief description here with an optional appendix if it needs to be expanded.

Author: added

### What is the high level implementation plan?

> Describe your plan on how to deliver this feature from prototyping to GA.
> Especially think about how to "bake" it in the open and get constant feedback
> from users before you stabilize the APIs.
>
> If you have a project board with your implementation plan, this is a good
> place to link to it.

### Are there any open issues that need to be addressed later?

> Describe any major open issues that this RFC did not take into account. Each

> > issue should have a GitHub issue associated with it and listed here.

Author: ACK

> ## Appendix: ....

Reviewer: I would add a few suggestive appendices, ex: 'changes required', 'alternatives considered', 'implementation plan' and maybe 'detailed customer experience'. 
These are all of course optional but would give the user a good template, and cover most RFCs.

Author: Added some example.

Reviewer: Suggestion: Similar to the working backwards section, I would prefer if we added a few appendix examples.

Author: We can add them later. I don't have a concrete list in my head and generally I don't want to encourage over-documentation.

Feel free to add any number of appendices as you see fit, but don't expect these

> to be reviewed. For example, you can add an appendix with a more detailed design

Reviewer: Why not expect these to be reviewed?

Author: Removed wording

---
rfc pr: [#xxx](https://github.com/aws/aws-cdk-rfcs/pull/xxx) <-- fill this after you've already created the PR
tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/yyy
---

# [TITLE]

> One sentence: brief description of the feature from a user perspective.

> ## Working Backwards

Reviewer: Besides these, we could consider an 'AWS What's New' post and blog post as good working backwards docs.

If you think so too, maybe mention it somewhere.

Author: What's new is the same as CHANGELOG, and a blog post is the same as README. At the moment I think those are sufficient.

> Describe use cases that are addressed by this feature.

## Internal FAQ

> The goal of this section is to help decide if this RFC should be implemented.
> It should include answers to questions that the team is likely ask. Contrary
> to the rest of the RFC, answers should be written "from the present" and
> likely discuss design approach, implementation plans, alternative considered
> and other considerations that will help decide if this RFC should be
> implemented.

### Why are we doing this?

> What is the motivation for this change?

### Why should we _not_ do this?

> Is there a way to address this use case with the current product? What are the
> downsides of implementing this feature?

### What changes are required to enable this change?

> Briefly describe the high-level design approach for implementing this feature.
>
> As appropriate, you can add an appendix with a more detailed design document.
>
> This is a good place to reference a prototype or proof of concept, which is
> highly recommended for most RFCs.

### Is this a breaking change?

> If the answer is no. Otherwise:
>
> Describe what ways did you consider to deliver this without breaking users?
>

> > Make sure to include a `BREAKING CHANGE` section above with a a description of the breaking

Reviewer: ```suggestion
> Make sure to include a `BREAKING CHANGE` clause under the CHANGELOG section with a description of the breaking
```

---
rfc pr: https://github.com/aws/aws-cdk-rfcs/pull/xxx <-- fill this after you've already created the PR
related issue: https://github.com/aws/aws-cdk-rfcs/issues/yyy
---

> One sentence: brief description of the feature in a user-facing terminology.

## Working Backwards

### README

> If this is a new feature, work backwards from the user-facing README artifact.

### RELEASE NOTES

> > If this change includes a major breaking change, add a "RELEASE NOTES" section with migration instructions

Reviewer 4: The term *RELEASE NOTES* is not usually coupled solely with breaking changes, but rather all changes. Maybe rename to *BREAKING CHANGES* to be more explicit?

Reviewer: @iliapolo - I believe this statement is meant to be additional, i.e., "write the release notes here. if this includes a major breaking change, this should include a section with migration instructions".

Maybe add one sentence in the beginning to briefly describe what goes here, i.e., "This section should contain a close representation on what a user would see in the release notes of the CDK when this feature is released"

Reviewer 4: @eladb Was the intent here to simulate our CHANGELOG entries?

Author: Refactored

> ### PRESS RELEASE

Reviewer 4: Not sure this will be clear to external contributors. We should add an example, or some instructions. But honestly i think we can even start without this. The release notes and readme should cover our working backwards strategy.

Author: Added details

> If this change is a major announcement, add a "PRESS RELEASE" section

## Frequently Asked Questions

> This is a list of ideas for questions that can help you think about various aspects of your proposal. 
> Pick the questions that help and add value and add any questions that you think might be useful. 
> No need to fill in arbitrary information if it's already covered in the previous section

### Why are we doing this? What is the motivation for this change?

### Why should we _not_ do this? What are the downsides?

### What alternative solutions did you consider?

> ### What changes are required to enable this change (high level design)?

Reviewer: I would rather have this as a separate section. This can range from 1 paragraph to several pages with its own sections, depending on the RFC. 

Would be nice to keep the 'design summary' and 'design details' sections from the existing RFC format.

Author: Added an option to add appendices

> ### Is this a breaking change? What migration path do we offer to users?

Reviewer: Already (should be, if not) covered in release notes.

Author: Cleaned that up to focus on the internal considerations and alternatives to breaking.

### Are there any open issues that need to be addressed later?

> ### Where I can find the implementation GitHub project board?

Reviewer 4: Does this imply that every RFC requires a project board for implementation?

Author: Changed to "What is the high level implementation plan"
